src/OrchardCore.Modules/OrchardCore.Twitter/Migrations/TwitterMigrations.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using Dapper;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Data;
+using OrchardCore.Data.Migration;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Environment.Shell.Descriptor.Models;
+using OrchardCore.Environment.Shell.Scope;
+using YesSql;
+
+namespace OrchardCore.Twitter.Migrations;
+
+public class TwitterMigrations : DataMigration
+{
+    private readonly ShellDescriptor _shellDescriptor;
+
+    public TwitterMigrations(ShellDescriptor shellDescriptor)
+    {
+        _shellDescriptor = shellDescriptor;
+    }
+
+    public int Create()
+    {
+        if (_shellDescriptor.WasFeatureAlreadyInstalled("OrchardCore.Twitter"))
+        {
+            ShellScope.AddDeferredTask(async scope =>
```

Comment:
Why is a deferred task necessary? This can be done just from within `Create`, migrations are for these kind of tasks. Then you don't need to manage the transaction either (what I don't think you need to do anyway).

Reply:
> Why is a deferred task necessary?

No reason other than it's what I could find from other PRs that looked to migrate data:
- https://github.com/OrchardCMS/OrchardCore/pull/11052/files#diff-74a3ba1fb26c512218a56465ffdfdc50b501eebab27ba6b14ae96161968e8214R152
- https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Search.Lucene/Migrations.cs#L178
- https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs#L252

> This can be done just from within Create

Sounds good - I'll look to move the migration code there.

Reply:
This can be necessary if you want to make sure that other migrations run before this. I don't think this is an issue here.

Reply:
> This can be necessary if you want to make sure that other migrations run before this. I don't think this is an issue here.

Ok, I refactored the code to remove the deferred task, and just update the database similar to what is found in the [documentation](https://docs.orchardcore.net/en/latest/reference/core/Data/).